### PR TITLE
fix: check for gateway backend online without http request

### LINKED ISSUE #10921: Calling metrics on gateway triggers unauthenticated calls to remote s3
I have a minio gateway and minio nas server. Since adding metrics via prometheus the number of 403's triggered on the nas instance has started to climb in synch with prometheus probes to the gateway.

## Expected Behavior

The metrics endpoint should be lightweight and not trigger side effects probably.
Calls from the gateway to the remote S3 should be authenticated.

## Current Behavior

mc admin trace on the storage mc instance shows this:

```
14:02:49.716 [403 Forbidden] s3.ListBuckets der2219.mgsops.net:9000/ 10.75.11.159      15.446ms     ↑ 31 B ↓ 483 B
14:02:49.892 [403 Forbidden] s3.ListBuckets der2219.mgsops.net:9000/ 10.75.11.159      992µs       ↑ 31 B ↓ 483 B
14:02:54.032 [403 Forbidden] s3.ListBuckets der2219.mgsops.net:9000/ 10.75.11.158      2ms          ↑ 31 B ↓ 483 B
14:02:54.209 [403 Forbidden] s3.ListBuckets der2219.mgsops.net:9000/ 10.75.11.158      994µs       ↑ 31 B ↓ 483 B
14:03:00.076 [403 Forbidden] s3.ListBuckets der2219.mgsops.net:9000/ 10.75.11.168      9.474ms      ↑ 31 B ↓ 483 B
14:03:00.255 [403 Forbidden] s3.ListBuckets der2219.mgsops.net:9000/ 10.75.11.168      3.003ms      ↑ 31 B ↓ 483 B
14:03:04.716 [403 Forbidden] s3.ListBuckets der2219.mgsops.net:9000/ 10.75.11.159      13.512ms     ↑ 31 B ↓ 483 B
```

## Steps to Reproduce (for bugs)

1. Save this `docker-compose.yml` file in a folder. It will create a subfolder ".mc" to store minio client credentials.

```yaml
version: "3.8"

services:
  minio:
    image: minio/minio:latest
    command: server /data
    environment:
      MINIO_ACCESS_KEY: access_key
      MINIO_SECRET_KEY: secret_key
  gateway:
    command: gateway s3 http://minio:9000
    image: minio/minio:latest
    environment:
      MINIO_PROMETHEUS_AUTH_TYPE: public
      MINIO_ACCESS_KEY: access_key
      MINIO_SECRET_KEY: secret_key
  mc:
    image: minio/mc
    volumes:
    - ./.mc:/root/.mc
  curl:
    image: curlimages/curl
```

2. execute the following commands from a bash or compatible terminal in the folder 

```bash
docker-compose up -d minio gateway
```

3. Run mc admin trace in one terminal to view the minio log:

```bash
docker-compose run mc alias set minio http://minio:9000 access_key secret_key
docker-compose run mc admin trace minio/
```

4. In another terminal, call the metrics endpoint on the gateway

```bash
docker-compose run curl http://gateway:9000/minio/prometheus/metrics
```

## Context

Its making the metrics noisy and difficult to track actual errors

## Your Environment

* Version used (`minio --version`): `minio version RELEASE.2020-11-13T20-10-18Z`

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
fix: check for gateway backend online without http request

## Motivation and Context
fixes #10921 

## How to test this PR?
Setup S3 gateway with Prometheus 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: cmd/gateway/azure/gateway-azure.go [hunk 1]
```diff
@@ -553,7 +553,11 @@ func (a *azureObjects) Shutdown(ctx context.Context) error {
 // StorageInfo - Not relevant to Azure backend.
 func (a *azureObjects) StorageInfo(ctx context.Context, _ bool) (si minio.StorageInfo, _ []error) {
 	si.Backend.Type = minio.BackendGateway
-	si.Backend.GatewayOnline = minio.IsBackendOnline(ctx, a.httpClient, a.endpoint)
+	host := a.endpoint.Host
+	if a.endpoint.Port() == "" {
+		host = a.endpoint.Host + ":" + a.endpoint.Scheme
```
Comment (vadmeste) [inline cmd/gateway/azure/gateway-azure.go:558]: Not sure why Scheme here

File: cmd/gateway/s3/gateway-s3.go [hunk 2]
```diff
@@ -276,7 +276,11 @@ func (l *s3Objects) Shutdown(ctx context.Context) error {
 // StorageInfo is not relevant to S3 backend.
 func (l *s3Objects) StorageInfo(ctx context.Context, _ bool) (si minio.StorageInfo, _ []error) {
 	si.Backend.Type = minio.BackendGateway
-	si.Backend.GatewayOnline = minio.IsBackendOnline(ctx, l.HTTPClient, l.Client.EndpointURL().String())
+	host := l.Client.EndpointURL().Host
+	if l.Client.EndpointURL().Port() == "" {
+		host = l.Client.EndpointURL().Host + ":" + l.Client.EndpointURL().Scheme
```
Comment (vadmeste) [inline cmd/gateway/s3/gateway-s3.go:281]: Not sure why Scheme here

**Review by vadmeste — COMMENTED** [review]
Some comments

File: cmd/gateway/azure/gateway-azure.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/gateway/azure/gateway-azure.go:558]: net.Dial needs to have port always @vadmeste

File: cmd/gateway/s3/gateway-s3.go (hunk 2 above)
Comment (harshavardhana) [reply cmd/gateway/s3/gateway-s3.go:281]: Same as above

File: cmd/gateway/azure/gateway-azure.go (hunk 1 above)
Comment (vadmeste) [reply cmd/gateway/azure/gateway-azure.go:558]: > net.Dial needs to have port always @vadmeste

yeah but a.endpoint.Scheme contains 'http'/'https' and not 80/443

Comment (harshavardhana) [reply cmd/gateway/azure/gateway-azure.go:558]: yes Go supports :http --> 80 and :https -> :443 @vadmeste  - https://golang.org/pkg/net/#Dial

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**